crates/miden-tx/src/executor/notes_checker.rs

```diff
@@ -127,6 +127,77 @@ where
+        // TODO: apply the static analysis before executing the tx
+
+        let tx_inputs = self
+            .0
+            .prepare_transaction_inputs(
+                account_id,
+                block_ref,
+                InputNotes::new_unchecked(vec![note]),
+                &tx_args,
+            )
+            .await
+            .map_err(NoteCheckerError::TransactionPreparation)?;
+
+        match self.try_execute_notes(&tx_inputs, &tx_args).await {
+            // execution succeeded
+            Ok(()) => Ok(NoteConsumabilitySignatureAware::Consumable),
+            Err(tx_checker_error) => {
+                match tx_checker_error {
+                    // execution failed on the preparation stage, before we actually executed the tx
+                    TransactionCheckerError::TransactionPreparation(e) => {
+                        Err(NoteCheckerError::TransactionPreparation(e))
+                    },
+                    // execution failed during the prologue
+                    TransactionCheckerError::PrologueExecution(e) => {
+                        Err(NoteCheckerError::PrologueExecution(e))
+                    },
+                    // execution failed during the note processing
+                    TransactionCheckerError::NoteExecution { .. } => {
+                        // TODO: apply the static analysis
+
+                        Ok(NoteConsumabilitySignatureAware::NonConsumable)
+                    },
+                    // execution failed during the epilogue
+                    // If we got the `Unauthorized` error, that means that the executor doesn't have
+                    // the signatures for the transaction authentication. In that case the
+                    // `ConsumableWithAuthorization` result is returned.
+                    // TODO: add additional epilogue checks, like fee assertions.
+                    TransactionCheckerError::EpilogueExecution(epilogue_error) => {
+                        match epilogue_error {
+                            // `Unauthorized` is returned for the multisig accounts if the
+                            // transaction doesn't have enough signatures.
+                            // `MissingAuthenticator` is returned for the account with the basic
+                            // auth if the authenticator was not provided to the executor
+                            // (UnreachableAuth).
+                            // Both these cases signal that there is a probability that the
+                            // note could be consumed if the authorization is provided.
+                            TransactionExecutorError::Unauthorized(_)
+                            | TransactionExecutorError::MissingAuthenticator => {
+                                Ok(NoteConsumabilitySignatureAware::ConsumableWithAuthorization)
+                            },
+                            _ => Ok(NoteConsumabilitySignatureAware::NonConsumable),
+                        }
```

Comment:
nit: maybe we should move this out into a helper function to make the code a bit more readable.